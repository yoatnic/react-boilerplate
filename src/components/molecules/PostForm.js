//@flow
import React from "react";

type Props = {
  onMessagePosted: Function
};

type State = {
  text: string
};

class PostForm extends React.Component<Props, State> {
  constructor() {
    super(...arguments);

    (this: any).handleClick = this.handleClick.bind(this);
    (this: any).handleChange = this.handleChange.bind(this);
    this.state = {
      text: ""
    };
  }

  handleClick() {
    if (!this.state.text) return;
    this.props.onMessagePosted(this.state.text);
    this.setState({ text: "" });
  }

  handleChange(e: SyntheticInputEvent<HTMLInputElement>) {
    this.setState({ text: e.target.value });
  }

  render() {
    return (
      <div>
        <input
          type="text"
          value={this.state.text}
          onChange={this.handleChange}
        />
        <button onClick={this.handleClick}>post message</button>
      </div>
    );
  }
}

export default PostForm;
