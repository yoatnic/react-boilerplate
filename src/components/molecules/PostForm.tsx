import * as React from "react";

interface Props {
  onMessagePosted: Function;
}

interface State {
  text: string;
}

class PostForm extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.handleClick = this.handleClick.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.state = {
      text: ""
    };
  }

  handleClick() {
    if (!this.state.text) return;
    this.props.onMessagePosted(this.state.text);
    this.setState({ text: "" });
  }

  handleChange(e: React.ChangeEvent<HTMLInputElement>) {
    this.setState({ text: e.target.value });
  }

  render() {
    return (
      <div>
        <input
          type="text"
          value={this.state.text}
          onChange={this.handleChange}
        />
        <button onClick={this.handleClick}>post message</button>
      </div>
    );
  }
}

export default PostForm;
