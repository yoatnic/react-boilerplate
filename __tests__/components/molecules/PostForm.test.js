import assert from "assert";
import React from "react";
import renderer from "react-test-renderer";
import { mount, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import sinon from "sinon";
import PostForm from "components/molecules/PostForm";

configure({ adapter: new Adapter() });

test("click button => post message and clear input field", () => {
  const postFunc = sinon.spy();
  const form = mount(<PostForm onMessagePosted={postFunc} />);

  form.find("input").simulate("change", {
    target: { value: "test message" }
  });
  form.find("button").simulate("click");

  assert(postFunc.calledWith("test message") === true);
  assert(postFunc.calledOnce === true);
  assert(form.find("input").props().value === "");
});

test("snapshot", () => {
  const tree = renderer
    .create(<PostForm onMessagePosted={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
