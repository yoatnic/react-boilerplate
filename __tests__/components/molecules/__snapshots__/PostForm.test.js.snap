// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snap shot 1`] = `
<div>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    post message
  </button>
</div>
`;
